src/runtimes/node/in_source_config/index.ts

```diff
@@ -37,22 +39,43 @@ const validateScheduleFunction = (functionFound: boolean, scheduleFound: boolean
 // Parses a JS/TS file and looks for in-source config declarations. It returns
 // an array of all declarations found, with `property` indicating the name of
 // the property and `data` its value.
-export const findISCDeclarationsInPath = async (sourcePath: string, functionName: string): Promise<ISCValues> => {
-  const ast = await safelyParseFile(sourcePath)
+export const findISCDeclarationsInPath = async (
+  sourcePath: string,
+  functionName: string,
+  featureFlags: FeatureFlags,
+): Promise<ISCValues> => {
+  const source = await safelyReadSource(sourcePath)
+
+  if (source === null) {
+    return {}
+  }
+
+  return findISCDeclarations(source, functionName, featureFlags)
+}
+
+export const findISCDeclarations = (source: string, functionName: string, featureFlags: FeatureFlags): ISCValues => {
+  const ast = parseSource(source)
 
   if (ast === null) {
     return {}
   }
 
   const imports = ast.body.flatMap((node) => getImports(node, IN_SOURCE_CONFIG_MODULE))
-
   const scheduledFunctionExpected = imports.some(({ imported }) => imported === 'schedule')
+
   let scheduledFunctionFound = false
   let scheduleFound = false
 
   const getAllBindings = createBindingsMethod(ast.body)
-  const mainExports = getMainExport(ast.body, getAllBindings)
-  const iscExports = mainExports
+  const { defaultExport, handlerExports } = getExports(ast.body, getAllBindings)
+
+  if (featureFlags.zisi_functions_api_v2 && handlerExports.length === 0 && defaultExport !== undefined) {
```

Comment:
Now the feature flag is used to control whether the detection mechanism kicks in. We should be able to roll it out safely to everyone and remove the flag entirely as a quick follow-up.